repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/versioning/VersioningForIdentityTest.groovy

```diff
@@ -0,0 +1,198 @@
+
+    def static Map contentXML = ["content-type": "application/xml"]
+    def static Map contentJSON = ["content-type": "application/json"]
+
+    //Start repose once for this particular translation test
+    def setupSpec() {
+        deproxy = new Deproxy()
+        deproxy.addEndpoint(properties.targetPort)
+        deproxy.addEndpoint(properties.targetPort2)
+
+        def params = properties.getDefaultTemplateParams()
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/versioning/identity", params)
+        repose.start()
+    }
+
+    def cleanupSpec() {
+        deproxy.shutdown()
+        repose.stop()
+    }
+
+    @Unroll
+    def "when retrieving all versions: #reqHeaders"() {
+        when: "User sends requests through repose"
+        def mc = deproxy.makeRequest(url: reposeEndpoint, method: 'GET', headers: reqHeaders)
+
+        then: "Response body should contain"
+        mc.receivedResponse.code == "200"
+
+        for (String st : shouldContain) {
+            mc.receivedResponse.body.contains(st)
+        }
+
+        where:
+        reqHeaders | shouldContain
+        acceptXML  | ['id="/v1"', 'id="/v2"']
+        acceptJSON | ['"id" : "/v1"', '"id" : "/v2"']
+    }
+
+    def "verify the response JSON is formatted for identity when the format is not configured for a versions request"() {
+        when: "User sends requests through repose"
+        def mc = deproxy.makeRequest(url: reposeEndpoint, method: 'GET', headers: acceptJSON)
+
+        then: "Response body should contain the expected JSON format"
+        mc.receivedResponse.code == "200"
+
+        def json = new JsonSlurper().parseText(mc.receivedResponse.body as String)
+        json.versions.values
+        json.versions.values.size == 5
+        json.versions.values[0].id
+        json.versions.values[0].id.contains("/v")
+        json.versions.values[0].links
+        json.versions.values[0].links.href
+        json.versions.values[0].links.rel
+        json.versions.values[0]."media-types"
+        json.versions.values[0]."media-types".base
+        json.versions.values[0]."media-types".base[0].contains("application/")
+        json.versions.values[0]."media-types".type
+        json.versions.values[0]."media-types".type[0].contains("application/")
+        json.versions.values[0].status
+        json.versions.values[0].status as String == "stable"
+        json.versions.values.find { it.id == '/v1' }.status == "deprecated"
+        json.versions.values.find { it.id == '/v2' }.status == "stable"
+        json.versions.values.find { it.id == '/v3' }.status == "deprecated"
+        json.versions.values.find { it.id == '/v4' }.status == "alpha"
+        json.versions.values.find { it.id == '/v5' }.status == "beta"
+    }
+
+    def "verify the response JSON is formatted for Identity when the format is not configured for a single version request"() {
```

Comment:
The method name should be updated to indicate that the format is being configured for the IDENTITY format.